Makefile

```diff
@@ -88,21 +88,22 @@ APPLICATION_VERSION_DEBIAN = $(shell echo $(APPLICATION_VERSION) | tr "-" "~")
 # Rules
 # ---------------------------------------------------------------------
 
-release/electron-$(TARGET_PLATFORM)-$(TARGET_ARCH)-app:
-	./scripts/unix/electron-create-resources-app.sh -s . -f "$(APPLICATION_FILES)" -o $@
+release/electron-$(TARGET_PLATFORM)-$(TARGET_ARCH)-dependencies/node_modules: package.json npm-shrinkwrap.json
+	./scripts/unix/dependencies-npm.sh -p -r "$(TARGET_ARCH)" -v "$(ELECTRON_VERSION)" -x $(dir $@) -t electron
 
-release/electron-$(TARGET_PLATFORM)-$(TARGET_ARCH)-app/node_modules:\
-	release/electron-$(TARGET_PLATFORM)-$(TARGET_ARCH)-app
-	./scripts/unix/dependencies-npm.sh -p -r "$(TARGET_ARCH)" -v "$(ELECTRON_VERSION)" -x $< -t electron
+release/electron-$(TARGET_PLATFORM)-$(TARGET_ARCH)-dependencies/bower_components: bower.json
+	./scripts/unix/dependencies-bower.sh -p -x $(dir $@)
 
-release/electron-$(TARGET_PLATFORM)-$(TARGET_ARCH)-app/bower_components:\
-	release/electron-$(TARGET_PLATFORM)-$(TARGET_ARCH)-app
-	./scripts/unix/dependencies-bower.sh -p -x $<
+release/electron-$(TARGET_PLATFORM)-$(TARGET_ARCH)-app: \
+	release/electron-$(TARGET_PLATFORM)-$(TARGET_ARCH)-dependencies/node_modules \
```

Comment:
Nearly. I'd do something more like:
```
release:
        mkdir $@

release/stamp_files: | release
        mkdir $@

release/electron-$(TARGET_PLATFORM)-$(TARGET_ARCH)-dependencies: | release
        mkdir $@

release/stamp_files/electron-$(TARGET_PLATFORM)-$(TARGET_ARCH)-npm-dependencies-installed.stamp: | release/stamp_files release/electron-$(TARGET_PLATFORM)-$(TARGET_ARCH)-dependencies
        ./scripts/unix/dependencies-npm.sh -p -r "$(TARGET_ARCH)" -v "$(ELECTRON_VERSION)" -x release/electron-$(TARGET_PLATFORM)-$(TARGET_ARCH)-dependencies -t electron
        touch $@
```
and then have other targets depending on `release/stamp_files/electron-$(TARGET_PLATFORM)-$(TARGET_ARCH)-npm-dependencies-installed.stamp`. So it's a stamp per Makefile-step, not a stamp per directory (and IMHO it makes sense to keep all the stamp-files in the same directory, so you can see at-a-glance what's been done). And then anywhere where you `rm -rf release/electron-$(TARGET_PLATFORM)-$(TARGET_ARCH)-dependencies/node_modules` you need to remember to `rm release/stamp_files/electron-$(TARGET_PLATFORM)-$(TARGET_ARCH)-dependencies-npm-installed.stamp` too (or create another Makefile rule to clean both in one step). Of course this can be all made to look **much** nicer with the use of Makefile variables (which is something I've started on, but haven't PRed yet, because I have to keep rebasing it! :wink: ).

The bit after the `|` in the rule is an order-only dependency, which means `make` only checks it exists, it doesn't compare timestamps.

In the project for my other client (which uses Perl rather than Bash and Javascript), I actually list the script as a dependency of the rule too, so that if the script gets modified then the rule gets re-run. That may or may not be appropriate here, depending on what each script is actually doing.